src/modules/crypto/xor.ts

```diff
@@ -0,0 +1,11 @@
+/** XOR two keys together, expects buffers to be of the same length */
+export const xor = (leftKey: Buffer, rightKey: Buffer): Buffer => {
+    if (leftKey.length !== rightKey.length) {
```

Comment:
can check the length here

`return Buffer.from(b1.map((b1, index) => b1 ^ b2[index]))`

Reply:
i'd like to keep it easy to read